src/UnitTests.Shared/DummyMappedDriveUtils.cs

```diff
@@ -0,0 +1,32 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#nullable enable
+using System;
+using Microsoft.Build.Framework;
+using Microsoft.Build.UnitTests.Shared;
+
+namespace Microsoft.Build.UnitTests.Shared;
+
+public static class DummyMappedDriveUtils
+{
+    public static DummyMappedDrive GetDummyMappedDrive(DummyMappedDrive mappedDrive)
```

Comment:
The ownership of `DummyMappedDrive` is confusing here.
How about just using `Lazy<DummyMappedDrive>`  by the calling code?
Or this util method can be left to return the Lazy. But it should not take mappedDrive to create mappedDrive.

Reply:
Thanks for your suggestions. Have updated.